Magento2/Sniffs/Legacy/AbstractBlockSniff.php

```diff
@@ -0,0 +1,90 @@
+<?php
+/**
+ * Copyright © Magento. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+namespace Magento2\Sniffs\Legacy;
+
+use PHP_CodeSniffer\Files\File;
+use PHP_CodeSniffer\Sniffs\Sniff;
+
+class AbstractBlockSniff implements Sniff
+{
+    const CHILD_HTML_METHOD = 'getChildHtml';
+    const CHILD_CHILD_HTML_METHOD = 'getChildChildHtml';
+
+    /**
+     * Warning violation code.
+     *
+     * @var string
+     */
+    protected $errorCode = 'FoundCountOfParametersIncorrect';
+
+    /**
+     * @inheritdoc
+     */
+    public function register(): array
+    {
+        return [
+            T_OBJECT_OPERATOR
+        ];
+    }
+
+    /**
+     * @inheritDoc
+     */
+    public function process(File $phpcsFile, $stackPtr)
+    {
+        if (!isset($phpcsFile->getTokens()[$stackPtr + 1]['content'])) {
+            return;
+        }
+        $content = $phpcsFile->getTokens()[$stackPtr + 1]['content'];
+        $isTheNextMethodGetChildSomething = in_array(
+            $content,
+            [
+                self::CHILD_HTML_METHOD,
+                self::CHILD_CHILD_HTML_METHOD
+            ]
+        );
+        if (!$isTheNextMethodGetChildSomething) {
+            return;
+        }
```

Comment:
Please extract this check to `isApplicable` private function
```suggestion
        if (!$this->isApplicable($content)) {
            return;
        }
```

Reply:
Done

Reply:
Thanks! Can you please also apply strict types to the class

Reply:
Also done!